client/src/store/Pipelines.ts

```diff
@@ -27,10 +18,10 @@ const pipelineModule: Module<PipelineState, never> = {
   actions: {
     async fetchPipelines({ commit, state }) {
       if (state.pipelines === null) {
-        const { data } = await getPipelineList() as {data: PipelineState};
+        const { data } = await getPipelineList();
```

Comment:
I think the issue you were having is that `PipelineState` has a key `pipelines` that isn't present in the actual json response.